### PR TITLE
Refactor route handlers

### LINKED ISSUE #27: Refactor Requests Handlers
They're in the style of 

```js
if ( method === "get" ) { 
    // handle
} else if (method === 'post') {
    // handle
}
```

We can split it up into multiple handler functions and get much better readability.

### PR DESCRIPTION
**Pull request by cadecairos**
fixes #27

### CODE REVIEW

File: services/api/handlers/users.js [hunk 1]
```diff
@@ -1,137 +1,142 @@
...
-module.exports = function(request, reply) {
-  if (request.method === 'get') {
-    return request.server.methods.db.findUser(
-      [request.params.user],
-      function(err, result) {
-        if ( err ) {
-          return reply(err);
-        }
+exports.get = function(request, reply) {
+  request.server.methods.db.findUser(
```
Comment (jbuck) [inline services/api/handlers/users.js]: Have you considered moving these server methods into the `pre` handlers for each route?

Comment (cadecairos) [inline services/api/handlers/users.js]: nope. Can you explain why we'd want that?

File: services/api/handlers/users.js [hunk 2]
```diff
@@ -1,137 +1,142 @@
...
+      reply({
+        status: 'created',
+        user: result.rows[0]
+      });
+    }
+  );
+};
+
+exports.patch = function(request, reply) {
+  request.server.methods.db.findUser( // TODO: promisify
```
Comment (jbuck) [inline services/api/handlers/users.js]: Can you file a follow-up issue for promisification?

Comment (cadecairos) [inline services/api/handlers/users.js]: filed #29

File: services/api/handlers/users.js (hunk 1 above)
Comment (jbuck) [inline services/api/handlers/users.js]: Sure! Take a look at the API here: http://hapijs.com/api#route-prerequisites

It's small, but I think you can reduce the repeated bit at the top in each of these handlers where you look up the user

**Comment by jbuck** [comment]
LGTM, r+

File: services/api/handlers/users.js (hunk 1 above)
Comment (cadecairos) [inline services/api/handlers/users.js]: Okay, I see what you mean. we can file that and do it separately.

**Pull request merged**